consensus/service/src/byzantine_ledger/metadata_provider.rs

```diff
@@ -37,10 +38,15 @@ impl<E: ReportableEnclave> ConsensusMetadataProvider<E> {
 
 impl<E: ReportableEnclave> BlockMetadataProvider for ConsensusMetadataProvider<E> {
     fn get_metadata(&self, block_data: &BlockData) -> Option<BlockMetadata> {
-        let verification_report = self
+        let attestation_evidence = self
             .enclave
             .get_attestation_evidence()
             .expect("failed to get AVR");
+        // TODO: replace with dcap
+        let verification_report = match attestation_evidence {
+            EvidenceKind::Epid(verification_report) => verification_report,
+            _ => panic!("Unreachable code"),
+        };
```

Comment:
Any reason why we can't already support both variants?
It looks like `AttestationEvidence` has them: https://github.com/mobilecoinfoundation/mobilecoin/blob/31eb5a0c92d8f12fca11abe6049d839e826c1d13/fog/report/types/src/lib.rs#L21

Reply:
TBH I'm not sure we need to have `EvidenceKind` here internally.
i.e. the nodes should only ever be talking with the same evidence kind. It seems like we only need to convert to `EvidenceKind` when leaving the nodes to external clients.

While here it can handle the `VerificationReport` other places can't so it seems nicer to prevent the invalid state/data type where it shouldn't be.  But I can also understand if we want to go into `EvidenceKind` as soon as possible.

Reply:
I see, that makes sense. So the idea is to have `get_attestation_evidence()` return `VerificationReport` until we switch to DCAP and then have it return `DcapEvidence`?
That seems like a good idea to me given that like you said it's only ever one or the other.

Reply:
That was my initial thought yes, this would stay verificationReport and then when dcap comes it becomes `DcapEvidence`. But that was me thinking more or less alone so didn't want to push too hard if people had other concerns/suggestions

Reply:
Changing this back to `VerificationReport` eventually makes it back [here](https://github.com/mobilecoinfoundation/mobilecoin/blob/4af1f8570535866bbb93d74b8420359672d99855/attest/ake/src/initiator.rs#L148). I'll change it back, but it's ultimately just moving the match somewhere else.

Reply:
Correct, the initiator.rs and its consumers will need to be evidencekind, but it limits the impact, especially of the `panic!(wrong evidence type)`, see this diff, https://github.com/mobilecoinfoundation/mobilecoin/compare/master...nick/multipledecode
I think the only place one needs to worry about `EvidenceKind` vs `DcapEvidence` in the enclave is in `peer_connect()`, https://github.com/mobilecoinfoundation/mobilecoin/compare/master...nick/multipledecode#diff-0ba700fa223a96139f81f0a74a737e8bc8c247ecdb6b1154a397a24faa2adde0R459